packages/frontend/src/components/ChartConfigPanel/Grid/index.tsx

```diff
@@ -14,42 +13,87 @@ export const defaultGrid: EchartsGrid = {
     bottom: '30', // pixels from bottom (makes room for x-axis)
 };
 
+const positions = ['top', 'bottom', 'left', 'right'] as const;
+
+type Units = '%' | 'px';
+
 const GridPanel: FC = () => {
     const {
         cartesianConfig: { dirtyEchartsConfig, setGrid },
     } = useVisualizationContext();
-    const methods = useForm<EchartsGrid>({
-        mode: 'onBlur',
-        defaultValues: { ...defaultGrid, ...dirtyEchartsConfig?.grid },
+
+    const [config, setConfig] = useState<EchartsGrid>({
```

Comment:
What do you think about turning this into a useMemo? that way we avoid the setGrid inside the setConfig and is always up to date with the context.

Reply:
@ZeRego good question. I initially implemented that way but its a bit slower because of the chart redraw I guess :thinkspin:  [](edited-with-Axolo)

Reply:
actually, it was useless so I removed it. thanks for the suggestion 👍